backends/p4tools/modules/smith/common/declarations.cpp

```diff
@@ -717,7 +717,10 @@ IR::ParameterList *DeclarationGenerator::genParameterList() {
     size_t numDirParams = (totalParams != 0U) ? Utils::getRandInt(0, totalParams - 1) : 0;
     size_t numDirectionlessParams = totalParams - numDirParams;
     for (size_t i = 0; i < numDirParams; i++) {
-        IR::Parameter *param = genTypedParameter(false);
+        // If the target is bmv2, generate a primitive-type parameter.
+        // Otherwise, generate a (normal) typed parameter (w/ more type options).
+        IR::Parameter *param =
```

Comment:
This is a good first approach but notice how you add target-specific information in the target-independent core. Not all tools know about BMv2 and there may be private back end. Another approach could be to override the virtual method in the BMv2 back end or parametrize it.